warehouse/query-core/src/main/java/datawave/query/data/parsers/EventKey.java

```diff
@@ -31,10 +32,18 @@ public class EventKey implements KeyParser {
     private String value;
     private String field;
 
+    private static final Logger log = Logger.getLogger(EventKey.class);
+
     @Override
     public void parse(Key k) {
-        clearState();
-        this.key = k;
+        if (this.key != null && this.key.equals(k)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Same key given to parse call. Skipping re-load: " + this.key.toStringNoTruncate());
+            }
+        } else {
+            clearState();
+            this.key = k;
```

Comment:
I don't know if this would ever occur - however, there are a few mutable methods on `Key`. Should/could the caller's `k` always get assigned to `key`? This comment would carry to the other `*Key` types too.